src/items.rs

```diff
@@ -45,12 +45,20 @@ impl Rewrite for ast::Local {
             let mut infix = String::new();
 
             if let Some(ref ty) = self.ty {
-                // 2 = ": ".len()
+                // 2 = ": ".len(), or 3 = " : ".len()
+                let len = if context.config.space_before_type_annotation {
+                    3
+                } else {
+                    2
+                };
+                let indent = offset + last_line_width(&result) + len;
                 // 1 = ;
-                let indent = offset + last_line_width(&result) + 2;
                 let budget = try_opt!(width.checked_sub(indent.width() + 1));
                 let rewrite = try_opt!(ty.rewrite(context, budget, indent));
 
+                if context.config.space_before_type_annotation {
+                    infix.push_str(" ");
+                }
                 infix.push_str(": ");
```

Comment:
I would pull this string out into a variable (with or without the extra space, then you can use `.len()` rather than the `len` variable above. That way you only need to check the option once.